src/prefect/utilities/importtools.py

```diff
@@ -26,6 +26,8 @@ def to_qualified_name(obj: Any) -> str:
     Returns:
         str: the qualified name
     """
+    if not hasattr(obj, "__qualname__"):
```

Comment:
Can we update this in `tasks/flows.py` to call `to_qualified_name(type(obj))` instead? `to_qualified_name` _should_ throw an error for instances because we will not be able to recreate it in `from_qualified_name`.

Reply:
Hmmm, I don't think that is a good idea because it will break existing behavior (?)
```
In [1]: def f():
   ...:     pass
   ...: 

In [2]: type(f).__qualname__
Out[2]: 'function'
```

Reply:
Ah, moving the logic to `Task.__init__`, something like:
```py
if not hasattr(obj, "__qualname__"):
    self.task_key = to_qualified_name(type(obj))
else:
    self.task_key = to_qualified_name(obj)
```

Reply:
Yep!